src/socket.rs

```diff
@@ -354,22 +351,17 @@ impl NlSocket {
     pub fn recv_ack(&mut self) -> Result<(), NlError> {
         if let Ok(ack) = self.recv_nl::<consts::Nlmsg, Nlmsgerr<consts::Nlmsg>>(None) {
             if ack.nl_type == consts::Nlmsg::Error && ack.nl_payload.error == 0 {
-                if self.pid.is_none() {
-                    self.pid = Some(ack.nl_pid);
-                } else if self.pid != Some(ack.nl_pid) {
-                    return Err(NlError::BadPid);
-                }
-                if let Some(seq) = self.seq {
-                    if seq != ack.nl_seq {
-                        return Err(NlError::BadSeq);
-                    }
-                }
```

Comment:
A lot of the code for PID and ACK handling is very broken and I think it's probably best to remove it for now until I get 0.5.0 released. I'm focused pretty heavily on doing it right there.